### PR TITLE
[DROOLS-7143] Drools 8 docs : Rule Unit DSL

### PR DESCRIPTION
**Pull request by tkobayas**
**JIRA**: 
https://issues.redhat.com/browse/DROOLS-7143

In this PR, I placed "Rule Unit DSL" section under "Build, Deploy, Utilize and Run" chapter. I'm not sure if this is the best place. I'm open for discussion and also we may rearrange the sections later.

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

- for <b>pull request checks</b>  
  Please add comment: <b>Jenkins retest this</b>

- for a <b>specific pull request check</b>  
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] tests</b>

- for a <b>full downstream build</b> 
  - for <b>jenkins</b> job: please add comment: <b>Jenkins run fdb</b>
  - for <b>github actions</b> job: add the label `run_fdb`

- <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

- <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

- <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>

- for <b>quarkus branch checks</b>  
  Run checks against Quarkus current used branch  
  Please add comment: <b>Jenkins run quarkus-branch</b>

- for a <b>quarkus branch specific check</b>  
  Run checks against Quarkus current used branch  
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] quarkus-branch</b>

- for <b>quarkus main checks</b>  
  Run checks against Quarkus main branch  
  Please add comment: <b>Jenkins run quarkus-main</b>

- for a <b>specific quarkus main check</b>  
  Run checks against Quarkus main branch  
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] quarkus-branch</b>

- for <b>native checks</b>  
  Run native checks  
  Please add comment: <b>Jenkins run native</b>

- for a <b>specific native check</b>  
  Run native checks 
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] native</b>

- for <b>mandrel checks</b>  
  Run native checks against Mandrel image
  Please add comment: <b>Jenkins run mandrel</b>

- for a <b>specific mandrel check</b>  
  Run native checks against Mandrel image  
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] mandrel</b>
</details>

### CODE REVIEW

File: drools-docs/src/modules/ROOT/pages/KIE/BuildDeployUtilizeAndRun/_rule-unit-dsl.adoc [hunk 1]
```diff
@@ -0,0 +1,73 @@
+[id='rule-unit-dsl_{context}']
+
+= Rule Unit DSL
+
+As introduced in xref:getting-started/index.adoc#first-rule-project_getting-started[First Rule Project], using Rule Units is a recommended style for implementing rules in {PRODUCT} 8. Detailed information in relation to DRL can be found in xref:language-reference/index.adoc#con-drl-rule-units_drl-rules[Rule units in DRL].
+
+However, {PRODUCT} 8 offers an alternative way of writing Rule Unit rules. Now you can write rules with Java APIs. Let's learn with comments on the example.
```
Comment (tarilabs) [inline drools-docs/src/modules/ROOT/pages/KIE/BuildDeployUtilizeAndRun/_rule-unit-dsl.adoc]:
```suggestion
Additionally, {PRODUCT} 8 offers an alternative way of writing rules in combination with Rule Unit. Now you can define rules for Rule Units using a dedicated set of Java APIs. Let's learn with comments on the example.
```

File: drools-docs/src/modules/ROOT/pages/KIE/BuildDeployUtilizeAndRun/_rule-unit-dsl.adoc [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+                          rule -> rule.on(ints) // creates a new pattern ...
+                                      .filter(GREATER_THAN, 5) // ... add an alpha filter to it
+                    ) // ... and join it with the former one
+                    .filter(EQUAL, String::length) // this filter is applied to the result of the join, so it is a beta constraint
+                    .execute(results, (r, s, i) -> r.add("String '" + s + "' is " + i + " characters long")); // the consequence captures all the joined variables positionally
+    }
+}
+----
+
+As you define rules using `defineRules` method, you can execute the rules, of course, without DRL.
```
Comment (tarilabs) [inline drools-docs/src/modules/ROOT/pages/KIE/BuildDeployUtilizeAndRun/_rule-unit-dsl.adoc]:
```suggestion
As you define rules using `defineRules` method, you can execute the rules, of course, without having to define them using DRL.
```

File: drools-docs/src/modules/ROOT/pages/KIE/index.adoc [hunk 3]
```diff
@@ -16,5 +16,6 @@ include::BuildDeployUtilizeAndRun/_executable-model-con.adoc[leveloffset=+1]
 include::BuildDeployUtilizeAndRun/_executable-model-modify-proc.adoc[leveloffset=+2]
 include::BuildDeployUtilizeAndRun/_executable-model-upgrading-proc.adoc[leveloffset=+2]
 include::BuildDeployUtilizeAndRun/_kie-scanner-proc.adoc[leveloffset=+1]
+include::BuildDeployUtilizeAndRun/_rule-unit-dsl.adoc[leveloffset=+1]
```
Comment (tarilabs) [inline drools-docs/src/modules/ROOT/pages/KIE/index.adoc]:
```suggestion
include::BuildDeployUtilizeAndRun/_rule-unit-dsl.adoc[leveloffset=+1]
include::BuildDeployUtilizeAndRun/_kie-scanner-proc.adoc[leveloffset=+1]
```

Imho is best earlier following the executable model sections

**Review by tarilabs — APPROVED** [review]
minor comments for your considerations (optional), thank you for the contribution!!

**Comment by tkobayas** [comment]
@tarilabs Thank you for the suggestions!

File: drools-docs/src/modules/ROOT/pages/KIE/BuildDeployUtilizeAndRun/_rule-unit-dsl.adoc [hunk 4]
```diff
@@ -0,0 +1,73 @@
+[id='rule-unit-dsl_{context}']
+
+= Rule Unit DSL
+
+As introduced in xref:getting-started/index.adoc#first-rule-project_getting-started[First Rule Project], using Rule Units is a recommended style for implementing rules in {PRODUCT} 8. Detailed information in relation to DRL can be found in xref:language-reference/index.adoc#con-drl-rule-units_drl-rules[Rule units in DRL].
+
+Additionally, {PRODUCT} 8 offers an alternative way of writing rules in combination with Rule Unit. Now you can define rules for Rule Units using a dedicated set of Java APIs. Let's learn with comments on the example.
```
Comment (kaldesai) [inline drools-docs/src/modules/ROOT/pages/KIE/BuildDeployUtilizeAndRun/_rule-unit-dsl.adoc]:
```suggestion
Additionally, {PRODUCT} 8 offers an alternative way of writing rules in combination with Rule Unit. Now you can define rules for Rule Units using a dedicated set of Java APIs. Let's learn with comments on the example.
```
```suggestion
Additionally, {PRODUCT} 8 offers an alternative way of writing rules in combination with Rule Unit. You can now define the rules for Rule Units using a dedicated set of Java APIs. Let's learn with comments on the example.
```

**Review by kaldesai — APPROVED** [review]
@tkobayas, Very minor suggestion. :) It looks good to me.  Thank you!!

**Comment by tkobayas** [comment]
Thank you, @kaldesai !

**Comment by tkobayas** [comment]
@mariofusco please review and merge if it's okay, thanks!

**Review by mariofusco — APPROVED** [review]

**Pull request merged**